apps/chat-e2e/src/tests/promptFoldersSpecialChars.test.ts

```diff
@@ -0,0 +1,225 @@
+
+    await dialTest.step('Rename it to contain special characters', async () => {
+      await folderPrompts.editFolderNameWithTick(newNameWithSpecialChars);
+      await expect
+        .soft(
+          folderPrompts.getFolderByName(newNameWithSpecialChars),
+          ExpectedMessages.folderNameUpdated,
+        )
+        .toBeVisible();
+    });
+
+    await dialTest.step(
+      'Try to type restricted special characters',
+      async () => {
+        await folderPrompts.openFolderDropdownMenu(newNameWithSpecialChars);
+        await folderDropdownMenu.selectMenuOption(MenuOptions.rename);
+        for (const char of ExpectedConstants.restrictedNameChars.split('')) {
+          await folderPrompts.editFolderName(char);
+          expect
+            .soft(
+              await folderPrompts.getEditFolderInput().getEditInputValue(),
+              ExpectedMessages.charactersAreNotDisplayed,
+            )
+            .toBe('');
+        }
+      },
+    );
+
+    await dialTest.step(
+      'Copy and paste restricted special characters',
+      async () => {
+        await page.evaluate(
+          (text) => navigator.clipboard.writeText(text),
+          nameWithRestrictedChars,
+        );
+        await folderPrompts.getEditFolderInput().editInput.click();
+        await page.keyboard.press(keys.ctrlPlusV);
+        await folderPrompts.getEditFolderInputActions().clickTickButton();
+        await expect
+          .soft(
+            folderPrompts.getFolderByName(expectedFolderName),
+            ExpectedMessages.folderIsVisible,
+          )
+          .toBeVisible();
+        await expect
+          .soft(
+            errorToast.getElementLocator(),
+            ExpectedMessages.noErrorToastIsShown,
+          )
+          .toBeHidden();
+      },
+    );
+
+    await dialTest.step(
+      'Rename folder to have spaces in the middle',
+      async () => {
+        await folderPrompts.openFolderDropdownMenu(expectedFolderName);
+        await folderDropdownMenu.selectMenuOption(MenuOptions.rename);
+        await folderPrompts.editFolderName(newNameWithSpaces);
+        // Assert that the input field displays the name with spaces
+        expect
+          .soft(
+            await folderPrompts.getEditFolderInput().getEditInputValue(),
+            ExpectedMessages.folderNameUpdated,
+          )
+          .toBe(newNameWithSpaces);
+        await folderPrompts.getEditFolderInputActions().clickTickButton();
+        // Assert that the folder is renamed with spaces on the panel
+        await expect
+          .soft(
+            folderPrompts.getFolderByName(newNameWithSpaces),
+            ExpectedMessages.folderNameUpdated,
+          )
+          .toBeVisible();
+      },
+    );
+
+    await dialTest.step(
+      'Prompt folder: name can not be blank or with spaces only',
+      async () => {
+        for (const name of ['', '   ']) {
+          await dialTest.step(`Try to rename folder to "${name}"`, async () => {
```

Comment:
to avoid lint warning, you may use loop inside dialTest.step():

`await dialTest.step('Prompt folder: name can not be blank or with spaces only',
      async () => {
        for (const name of ['', '   ']) {
....`

Reply:
fixed

Reply:
you don't need step inside step: 

`await dialTest.step(
        'Prompt folder: name can not be blank or with spaces only',
        async () => {
            for (const name of ['', '   ']) {
              await folderPrompts.openFolderDropdownMenu(newNameWithSpaces);
              await folderDropdownMenu.selectMenuOption(MenuOptions.rename);
              await folderPrompts.editFolderName(name);
              await folderPrompts.getEditFolderInputActions().clickTickButton();
              await promptBarFolderAssertion.assertFolderState(
                { name: newNameWithSpaces },
                'visible',
              );
            }
        },
      );`